deployer/cloudflaredeployer.go

```diff
@@ -0,0 +1,312 @@
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/pkg/errors"
+
+	"github.com/suborbital/atmo/atmo/appsource"
+	"github.com/suborbital/atmo/atmo/options"
+	"github.com/suborbital/atmo/directive"
+	subotemplate "github.com/suborbital/subo/builder/template"
+	"github.com/suborbital/subo/project"
+	"github.com/suborbital/subo/subo/util"
+)
+
+const (
+	cloudflareDeployJobType = "cloudflare"
+	routerScriptName        = "subo-router"
+
+	bundlePath = "./runnables.wasm.zip"
+)
+
+//go:embed cloudflare/shared.js
+var shared string
+
+//go:embed cloudflare/itty-router.js
+var ittyRouter string
+
+type CloudflareDeployJob struct {
+	domain          string
+	repo            string
+	branch          string
+	updateTemplates bool
+}
+
+func NewCloudflareDeployer(repo, branch, domain string, updateTemplates bool) DeployJob {
+	c := &CloudflareDeployJob{
+		repo:            repo,
+		branch:          branch,
+		domain:          domain,
+		updateTemplates: updateTemplates,
+	}
+	return c
+}
+
+func (k *CloudflareDeployJob) Type() string {
+	return cloudflareDeployJobType
+}
+
+type cfWorker struct {
+	Path           string
+	Method         string
+	ServiceBinding string
+}
+
+func (job *CloudflareDeployJob) Deploy(log util.FriendlyLogger, ctx *project.Context) error {
+	atmoOpts := options.NewWithModifiers()
+	appSource := appsource.NewBundleSource(bundlePath)
+
+	if err := appSource.Start(*atmoOpts); err != nil {
+		return errors.Wrap(err, "failed to start")
+	}
+
+	api, err := getAPIClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to initalize Cloudflare api client")
+	}
+
+	if err := os.RemoveAll(filepath.Join(ctx.Cwd, ".deployment")); err != nil {
+		if !os.IsNotExist(err) {
+			return errors.Wrap(err, "failed to RemoveAll deployment files")
+		}
+	}
+
+	if err := os.MkdirAll(filepath.Join(ctx.Cwd, ".deployment"), util.PermDirectory); err != nil {
+		return errors.Wrap(err, "failed to MkdirAll .deployment")
+	}
+
+	templatesPath, err := subotemplate.FullPath(job.repo, job.branch)
+	if err != nil {
+		return errors.Wrap(err, "failed to template.FullPath")
+	}
+
+	if job.updateTemplates {
+		templatesPath, err = subotemplate.UpdateTemplates(job.repo, job.branch)
+		if err != nil {
+			return errors.Wrap(err, "🚫 failed to UpdateTemplates")
+		}
+	}
+
+	for _, meta := range appSource.Applications() {
+		log.LogStart(fmt.Sprintf("deploying %s (%s) runnables as workers", meta.Identifier, job.domain))
+
+		runnablesByName := getRunnablesByName(appSource, meta)
+
+		workersToRoute := make(map[string]cfWorker)
+
+		for _, handler := range appSource.Handlers(meta.Identifier, meta.AppVersion) {
+			if handler.Input.Type != "request" {
+				log.LogWarn(fmt.Sprintf("unsupported handler type: %s; ignoring.", handler.Input.Type))
+				continue
+			}
+
+			if len(handler.Steps) != 1 {
+				log.LogWarn(fmt.Sprintf("only one step is currently supported per handler"))
```

Comment:
I'd like to include information about which handler has more than 1 step, and I think this should be an error that halts the deployment rather than a warning and `continue`